### PR TITLE
Warning if two level aggregation is disabled in distributed query

### PR DESCRIPTION
**Pull request by devcrafter**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Warning if two level aggregation is disabled in distributed query due to version incompatibility between initiator and executor servers

### CODE REVIEW

**Review by antonio2368 — APPROVED** [review]

**Review by KochetovNicolai — APPROVED** [review]

File: src/Interpreters/InterpreterSelectQuery.cpp [hunk 1]
```diff
@@ -2911,16 +2911,29 @@ void InterpreterSelectQuery::initSettings()
         InterpreterSetQuery(query.settings(), context).executeForCurrentContext();
 
     auto & client_info = context->getClientInfo();
-    auto min_major = DBMS_MIN_MAJOR_VERSION_WITH_CURRENT_AGGREGATION_VARIANT_SELECTION_METHOD;
-    auto min_minor = DBMS_MIN_MINOR_VERSION_WITH_CURRENT_AGGREGATION_VARIANT_SELECTION_METHOD;
+    const auto min_major = DBMS_MIN_MAJOR_VERSION_WITH_CURRENT_AGGREGATION_VARIANT_SELECTION_METHOD;
+    const auto min_minor = DBMS_MIN_MINOR_VERSION_WITH_CURRENT_AGGREGATION_VARIANT_SELECTION_METHOD;
```
Comment (antonio2368) [inline src/Interpreters/InterpreterSelectQuery.cpp:2915]: These can actually be `static constexpr`.

**Review by antonio2368 — COMMENTED** [review]
@devcrafter why isn't this merged?

**Review by alexey-milovidov — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
There is something interesting with the Replicated database engine.

**Comment by alexey-milovidov** [comment]
Tests are still failing.

**Pull request closed without merging**